yjit/src/codegen.rs

```diff
@@ -6014,6 +6002,23 @@ fn push_splat_args(required_args: u32, asm: &mut Assembler) {
     }
 }
 
+c_callable! {
+    /// Return non-zero when `obj` is an array and its last item is a
+    /// `ruby2_keywords` hash. We don't support this kind of splat.
+    fn splat_array_ends_in_ruby2_keywords(obj: VALUE) -> usize {
+        unsafe {
+            if !RB_TYPE_P(obj, RUBY_T_ARRAY) {
+                return 0;
+            }
+            let last = rb_ary_entry_internal(obj, -1);
+            if !RB_TYPE_P(last, RUBY_T_HASH) {
```

Comment:
This `RB_TYPE_P` is defined in yjit.c, so it's a couple of C function calls. Since it's used for rest & splat calls at runtime, can we either import the `RB_TYPE_P` definition to Rust or just write this function in C?

Reply:
Same applies to `rb_ary_entry_internal`.